src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -22,8 +23,37 @@
 
 class OrchestratorCli(orchestrator.OrchestratorClientMixin, MgrModule):
     MODULE_OPTIONS = [
-        {'name': 'orchestrator'}
+        {
+            'name': 'orchestrator',
+            'default': None,
+            'desc': 'Orchestrator backend',
+            'enum_allowed': ['ssh', 'rook', 'ansible', 'deepsea'],
+            'runtime': True,
+        },
+        {
+            'name': 'container_image_base',
```

Comment:
Guess we have to decide now.

Either we make the orchestrator independent of the deployment tool, or we make the orchestrator hard-coded for ssh and rook. 

I'm torn. If we restrict ourselves to ssh and rook, it will make things easier. On the other hand, will might alienate people that want to maintain mgr/ansible in the future. And it makes some of the compromises somewhat redundant that we made in the past.

Reply:
I can push the version magic down into mgr/ssh, so that (at this layer) it's an opaque string description of the version... that will kick the can a bit further down the road.

Reply:
Wondering. For Rook, this should be equal to the CephCluster CR's `spec.cephVersion.image` attribute. So, we can just query Rook instead of saving this here for all orchestrators?

Reply:
I think spec.CephVersion.image is the full image name + tag; the idea here is that this is the image without a tag, so that we can add vX.Y.Z to it.

Reply:
Getting late here in Europe, but I'm starting to get the impression that we're really overcomplicating things here. I'm going to rethink this tomorrow